.github/workflows/check-replays.yml

```diff
@@ -0,0 +1,236 @@
+          # BINKW32.DLL
+          # English.big
+          # INI.big
+          # Maps.big
+          # mss32.dll
+          # W3D.big
+          # Data\Scripts\MultiplayerScripts.scb
+          # Data\Scripts\SkirmishScripts.scb
+
+          Write-Host "Downloading GameData Generals" -ForegroundColor Cyan
+          aws s3 cp s3://github-ci/generals108_gamedata_trimmed.7z generals108_gamedata_trimmed.7z --endpoint-url $env:AWS_ENDPOINT_URL
+
+          Write-Host "Verifying File Integrity" -ForegroundColor Cyan
+          $fileHash = (Get-FileHash -Path generals108_gamedata_trimmed.7z -Algorithm SHA256).Hash
+          Write-Host "Downloaded file SHA256: $fileHash"
+          Write-Host "Expected file SHA256: $env:EXPECTED_HASH_GENERALS"
+          if ($fileHash -ne $env:EXPECTED_HASH_GENERALS) {
+              Write-Error "Hash verification failed! File may be corrupted or tampered with."
+              exit 1
+          }
+
+          Write-Host "Extracting Archive" -ForegroundColor Cyan
+          & 7z x generals108_gamedata_trimmed.7z -oC:\GameData\Generals
+          Remove-Item generals108_gamedata_trimmed.7z -Verbose
+
+          # Download GeneralsMD (ZH) Game Files
+          # The archive contains these files:
+          # BINKW32.DLL
+          # INIZH.big
+          # MapsZH.big
+          # mss32.dll
+          # W3DZH.big
+          # Data\Scripts\MultiplayerScripts.scb
+          # Data\Scripts\Scripts.ini
+          # Data\Scripts\SkirmishScripts.scb
+
+          Write-Host "Downloading GameData GeneralsMD" -ForegroundColor Cyan
+          aws s3 cp s3://github-ci/zerohour104_gamedata_trimmed.7z zerohour104_gamedata_trimmed.7z --endpoint-url $env:AWS_ENDPOINT_URL
+
+          Write-Host "Verifying File Integrity" -ForegroundColor Cyan
+          $fileHash = (Get-FileHash -Path zerohour104_gamedata_trimmed.7z -Algorithm SHA256).Hash
+          Write-Host "Downloaded file SHA256: $fileHash"
+          Write-Host "Expected file SHA256: $env:EXPECTED_HASH_GENERALSMD"
+          if ($fileHash -ne $env:EXPECTED_HASH_GENERALSMD) {
+              Write-Error "Hash verification failed! File may be corrupted or tampered with."
+              exit 1
+          }
+
+          Write-Host "Extracting Archive" -ForegroundColor Cyan
+          & 7z x zerohour104_gamedata_trimmed.7z -oC:\GameData\GeneralsMD
+          Remove-Item zerohour104_gamedata_trimmed.7z -Verbose
+
+      - name: Set Up Game Data
+        shell: pwsh
+        run: |
+          $source = "C:\GameData\${{ inputs.game }}"
+          $destination = "build"
+          Copy-Item -Path $source\* -Destination $destination -Recurse -Force
+
+      - name: Set Generals InstallPath in Registry
+        shell: pwsh
+        run: |
+          # Zero Hour loads some Generals files and needs this registry key to find the
+          # Generals data files.
+
+          $regPath = "HKCU:\SOFTWARE\Electronic Arts\EA Games\Generals"
+          $installPath = "C:\GameData\Generals\"
```

Comment:
This script maintains multiple variables with the same strings:

1. `path: C:\GameData`
2. `-oC:\GameData\Generals`
3. `$installPath = "C:\GameData\Generals\"`
4. `$source = "C:\GameData\${{ inputs.game }}"`

I think it would be good to share the names.

So for example

```
$gamePath = "C:\GameData"
$generalsPath = "${{gamePath}}\Generals"
$generalsMDPath = "${{gamePath}}\GeneralsMD"
```

Then it will be easier to tweak locations without breaking things.